### PR TITLE
Fix a couple of warnings.

### PR DESCRIPTION
**Pull request by rpdigos**
* **Addressed Issue(s)**: None

* **Server Mode**: Pre-Renewal and Renewal

* **Description of Pull Request**: 
   * Fix a couple of warnings on gcc

### CODE REVIEW

**Comment by rpdigos** [comment]
Error messages from GCC (4.8.5) covered by this pull request:

> clif.cpp:940:10: warning: 10 enumeration values not handled in switch: 'BL_NUL',
      'BL_PC', 'BL_MOB'... [-Wswitch]
        switch( bl->type ) {
                ^
clif.cpp:940:10: note: add missing switch cases
        switch( bl->type ) {
                ^
clif.cpp:1437:10: warning: 8 enumeration values not handled in switch: 'BL_NUL',
      'BL_HOM', 'BL_MER'... [-Wswitch]
        switch (bl->type)
                ^
clif.cpp:1437:10: note: add missing switch cases
        switch (bl->type)
                ^
clif.cpp:1720:9: warning: 9 enumeration values not handled in switch: 'BL_NUL',
      'BL_HOM', 'BL_MER'... [-Wswitch]
        switch(bl->type) {
               ^
clif.cpp:1720:9: note: add missing switch cases
        switch(bl->type) {
               ^
clif.cpp:4633:10: warning: 7 enumeration values not handled in switch: 'BL_NUL',
      'BL_HOM', 'BL_ITEM'... [-Wswitch]
        switch (bl->type)
                ^
clif.cpp:4633:10: note: add missing switch cases
        switch (bl->type)
                ^
clif.cpp:8042:12: warning: 10 enumeration values not handled in switch:
      'BL_NUL', 'BL_MOB', 'BL_PET'... [-Wswitch]
    switch(bl->type){
           ^
clif.cpp:8042:12: note: add missing switch cases
    switch(bl->type){
           ^
clif.cpp:11492:10: warning: 9 enumeration values not handled in switch:
      'BL_NUL', 'BL_PET', 'BL_HOM'... [-Wswitch]
        switch (bl->type) {
                ^
clif.cpp:11492:10: note: add missing switch cases
        switch (bl->type) {
                ^
7 warnings generated.
	CXX	map.cpp
map.cpp:2652:9: warning: 8 enumeration values not handled in switch: 'BL_NUL',
      'BL_PC', 'BL_PET'... [-Wswitch]
        switch(bl->type) {
               ^
map.cpp:2652:9: note: add missing switch cases
        switch(bl->type) {
               ^
map.cpp:4266:9: warning: 6 enumeration values not handled in switch: 'BL_NUL',
      'BL_HOM', 'BL_MER'... [-Wswitch]
        switch(bl->type) {
               ^
map.cpp:4266:9: note: add missing switch cases
        switch(bl->type) {
               ^
2 warnings generated.
	CXX	script.cpp
script.cpp:17446:10: warning: 6 enumeration values not handled in switch:
      'BL_NUL', 'BL_PC', 'BL_ITEM'... [-Wswitch]
        switch (bl->type) {
                ^
script.cpp:17446:10: note: add missing switch cases
        switch (bl->type) {
                ^
script.cpp:18149:10: warning: 8 enumeration values not handled in switch:
      'BL_NUL', 'BL_PC', 'BL_MOB'... [-Wswitch]
        switch (bl->type) {
                ^
script.cpp:18149:10: note: add missing switch cases
        switch (bl->type) {
                ^
script.cpp:18205:10: warning: 9 enumeration values not handled in switch:
      'BL_NUL', 'BL_PC', 'BL_MER'... [-Wswitch]
        switch (bl->type) {
                ^
script.cpp:18205:10: note: add missing switch cases
        switch (bl->type) {
                ^
script.cpp:19929:9: warning: enumeration value 'IM_MAX' not handled in switch
      [-Wswitch]
        switch(instance_data[instance_id].mode) {
               ^
script.cpp:19929:9: note: add missing switch cases
        switch(instance_data[instance_id].mode) {
               ^
script.cpp:20657:11: warning: address of 'sd->sc' will always evaluate to 'true'
      [-Wpointer-bool-conversion]
        if( &sd->sc && sd->sc.data[SC_ALL_RIDING] )
             ~~~~^~ ~~
script.cpp:20675:11: warning: address of 'sd->sc' will always evaluate to 'true'
      [-Wpointer-bool-conversion]
        if( &sd->sc && sd->sc.option&(OPTION_WUGRIDER|OPTION_RIDING|OPTI...
             ~~~~^~ ~~
script.cpp:20679:12: warning: address of 'sd->sc' will always evaluate to 'true'
      [-Wpointer-bool-conversion]
                if( &sd->sc && sd->sc.data[SC_ALL_RIDING] )
                     ~~~~^~ ~~

File: src/map/script.cpp [hunk 1]
```diff
@@ -19928,6 +19937,7 @@ static int buildin_instance_warpall_sub(struct block_list *bl, va_list ap)
 	owner_id = instance_data[instance_id].owner_id;
 	switch(instance_data[instance_id].mode) {
 		case IM_NONE:
+		case IM_MAX:
```
Comment (Jeybla) [inline src/map/script.cpp]: `IM_MAX` is afaik not a real instance mode. So it should fail in this case. Personally I'd just remove this case and leave it as it is.

**Comment by anacondaqq** [comment]
This is a step back, not forward. Use modern g++ /clang.

**Comment by Jeybla** [comment]
@anacondaqq mostly yes, but the removal from the `&sd->sc &&` checks are useful. Anyway thanks @rpdigos for contributing! But most of the changes aren't really necessary.

**Comment by rpdigos** [comment]
@Jeybla @anacondaqq
No problem, just implement what you think useful. Thanks for the feedback.

**Comment by anacondaqq** [comment]
@rpdigos hi, sorry, don't mean my message offensive. The problem what you explain can be solved by upgrading compiler. Your fix, of course, can be applied, but what will happen next, when clif.c will be changed, then other .c files in map servers and so on? Devs already switched to C++ project and trying to do their best to make everything work and they officially declare requirements for the compiler (version > 5). Where is a sense to make work code on an outdated compiler which will provoke only problems in future, while all of these problems can be solved via two commands in terminal (by upgrading compiler, or switching to a clang)?

**Comment by rpdigos** [comment]
@anacondaqq 
I've found out that g++ on macOS Sierra is like an alias to:
Apple LLVM version 9.0.0 (clang-900.0.37)
Target: x86_64-apple-darwin16.7.0

So, to update my first post, these warnings in cpp files was not generated on gcc 4.5 but in the compiler above (clang)

I've installed gcc5 but it fails to compile with many undefined symbols like :
> Undefined symbols for architecture x86_64:
  "std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::find_last_not_of(char, unsigned long) const", referenced from:
      YAML::ScanScalar[abi:cxx11](YAML::Stream&, YAML::ScanScalarParams&) in yaml-cpp.a(scanscalar.o)

Seems it's related to https://stackoverflow.com/questions/33394934/converting-std-cxx11string-to-stdstring

This is just to inform. I'll try to fix my enviroment later. Thanks

**Pull request merged**